### PR TITLE
Added the dev script setup

### LINKED ISSUE #29: Use a pip .venv instead of pipenv
Acceptance criteria:
* check that `dls_dev_env.sh` runs okay
* check tests pass

### PR DESCRIPTION
**Pull request by evalott100**
Fixes #29

@DominicOram I added the dev environment script, This comes with a few warnings (e.g it wants a pyproject.toml and Distutils is imported before setup tools). Code tests pass, and tickit devices run fine. Idk if you want to fix the warnings here or make it a separate PR?

### CODE REVIEW

File: .github/workflows/code.yml [hunk 1]
```diff
@@ -2,45 +2,63 @@ name: Code CI
...
-    branches:
-      # Restricting to these branches and tags stops duplicate jobs on internal
-      # PRs but stops CI running on internal branches without a PR. Delete the
-      # next 5 lines to restore the original behaviour
-      - master
-      - main
   pull_request:
+  schedule:
+    # Run every Monday at 8am to check latest versions of dependencies
+    - cron: "0 8 * * WED"
```
Comment (DominicOram) [inline .github/workflows/code.yml:8]: Should: This comment disagrees with the cron job

**Review by DominicOram — CHANGES_REQUESTED** [review]
This looks good, can we:
* Add the pre-commit yaml as in https://github.com/dls-controls/tickit-devices/pull/24/files?
* Install the hooks in the dev_env script
* Update the README

**Comment by abbiemery** [comment]
@evalott100 This may be solved now by #32

**Comment by evalott100** [comment]
@abbiemery Yep, all it would be adding is the .sh to do the pip venv installation which I think is unnecessary anyway. Closing.

**Pull request closed without merging**